source/common/stats/thread_local_store.cc

```diff
@@ -347,6 +350,10 @@ void ParentHistogramImpl::merge() {
   std::unique_lock<std::mutex> lock(merge_lock_);
   if (usedLockHeld()) {
     hist_clear(interval_histogram_);
+    // Here we could copy all the pointers to TLS histograms in the tls_histogram_ list,
+    // then release the lock before we do the actual merge. However it is not a big deal
+    // because the tls_histogram merge is not that expensive as it is a single histogram
+    // merge and adding TLS histograms is rare.
```

Comment:
OK, sounds good.  One thing to look out for in the future is contention with used(), but I don't have a feel for how often that gets called.